// tailwindcss/lib/generators/sizing.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

exports.default = function (config) {
  return _lodash2.default.flatten([defineWidths(config.width), defineMinWidths(config.minWidth), defineMaxWidths(config.maxWidth), defineHeights(config.height), defineMinHeights(config.minHeight), defineMaxHeights(config.maxHeight)]);
};

var _lodash = require('lodash');

var _lodash2 = _interopRequireDefault(_lodash);

var _defineClass = require('../util/defineClass');

var _defineClass2 = _interopRequireDefault(_defineClass);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function defineWidths(widths) {
  return _lodash2.default.map(widths, (size, modifer) => {
    return (0, _defineClass2.default)(`w-${modifer}`, {
      width: `${size}`
    });
  });
}

function defineMinWidths(widths) {
  return _lodash2.default.map(widths, (size, modifer) => {
    return (0, _defineClass2.default)(`min-w-${modifer}`, {
      'min-width': `${size}`
    });
  });
}

function defineMaxWidths(widths) {
  return _lodash2.default.map(widths, (size, modifer) => {
    return (0, _defineClass2.default)(`max-w-${modifer}`, {
      'max-width': `${size}`
    });
  });
}

function defineHeights(heights) {
  return _lodash2.default.map(heights, (size, modifer) => {
    return (0, _defineClass2.default)(`h-${modifer}`, {
      height: `${size}`
    });
  });
}

function defineMinHeights(heights) {
  return _lodash2.default.map(heights, (size, modifer) => {
    return (0, _defineClass2.default)(`min-h-${modifer}`, {
      'min-height': `${size}`
    });
  });
}

function defineMaxHeights(heights) {
  return _lodash2.default.map(heights, (size, modifer) => {
    return (0, _defineClass2.default)(`max-h-${modifer}`, {
      'max-height': `${size}`
    });
  });
}
